src/adr/0040-generalize-event-bus.md

```diff
@@ -9,18 +9,15 @@ Proposed
 ## Context
 
 When searching for users by quota ratio if we set the value of the parameters to 0, the search feature is supposed to return newly created users
-who have not receive any email yet at that point. However, this is not the case due to the quotas are currently being initialize only after
+who have not received any email yet at that point. However, this is not the case due to the quotas are currently being initialized only after
 an user receive the first email.
 
-In order to solve this problem, we need to have some kind of event-driven system so when an new user is created successfully, 
-their quota should also be created immediately after. 
-
-The EventBus components currently implemented in mailbox-api would be suitable for this purpose.
+In order to solve this problem, we need to initialize user quotas on creation time. Event driven system allows indirection so that the users api 
+don't need a dependency on the quota components. The EventBus components currently defined in in mailbox-api would be suitable for this purpose.
 
 ## Decision
 
-Move the EventBus outside of mailbox-api so it can be implemented anywhere in James by adding the corresponding dependency. 
-One potential location is the server module.
+Extract the EventBus out of mailbox-api in order to make it a utility component so it can be used by all other components.
```

Comment:
I propose to add in the decision something that solves the initial concern.

Could you detail what you would do to solve that bug once you can use the EventBus for it?